src/preferences/dialog/dlgprefmetadatadlg.ui

```diff
@@ -0,0 +1,175 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <class>DlgPrefMetadataDlg</class>
+ <widget class="QWidget" name="DlgPrefMetadataDlg">
+  <property name="geometry">
+   <rect>
+    <x>0</x>
+    <y>0</y>
+    <width>668</width>
+    <height>514</height>
+   </rect>
+  </property>
+  <property name="windowTitle">
+   <string>Metadata Broadcast Preferences</string>
+  </property>
+  <layout class="QVBoxLayout" name="verticalLayout">
+   <item>
+    <widget class="QGroupBox" name="fileListenerBox">
+     <property name="title">
+      <string>Metadata file options</string>
+     </property>
+     <layout class="QVBoxLayout" name="verticalLayout_2">
+      <item>
+       <widget class="QCheckBox" name="enableFileListener">
+        <property name="text">
+         <string>Enable metadata file</string>
+        </property>
+       </widget>
+      </item>
+      <item>
+       <layout class="QHBoxLayout" name="horizontalLayout">
+        <item>
+         <widget class="QLabel" name="label_2">
+          <property name="text">
+           <string>File encoding</string>
+          </property>
+         </widget>
+        </item>
+        <item>
+         <spacer name="horizontalSpacer">
+          <property name="orientation">
+           <enum>Qt::Horizontal</enum>
+          </property>
+          <property name="sizeType">
+           <enum>QSizePolicy::Expanding</enum>
+          </property>
+          <property name="sizeHint" stdset="0">
+           <size>
+            <width>40</width>
+            <height>20</height>
+           </size>
+          </property>
+         </spacer>
+        </item>
+        <item>
+         <widget class="QComboBox" name="fileEncodingComboBox">
+          <property name="enabled">
+           <bool>false</bool>
+          </property>
+          <property name="sizePolicy">
+           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
+            <horstretch>0</horstretch>
+            <verstretch>0</verstretch>
+           </sizepolicy>
+          </property>
+          <property name="minimumSize">
+           <size>
+            <width>200</width>
+            <height>0</height>
+           </size>
+          </property>
+          <property name="editable">
+           <bool>false</bool>
+          </property>
+          <property name="insertPolicy">
+           <enum>QComboBox::NoInsert</enum>
+          </property>
+          <property name="sizeAdjustPolicy">
+           <enum>QComboBox::AdjustToContents</enum>
+          </property>
+         </widget>
+        </item>
+       </layout>
+      </item>
+      <item>
+       <widget class="QLabel" name="label">
+        <property name="text">
+         <string>File format. The strings {{ track.artist }} and {{ track.title }} will be substituted by the metadata of the current track.</string>
```

Comment:
Are there any more fields available? If not, should we make them available?

Reply:
Might be a good idea to already file a PR for the mixxx manual that documents the fields that are available.

Reply:
There is currently no demand for this. If we have a real request, we can add any info we have.

Reply:
> Might be a good idea to already file a PR for the mixxx manual that documents the fields that are available.

Can you point me to the place where I can add it?

Reply:
> Are there any more fields available?

https://github.com/mixxxdj/manual/blob/main/source/chapters/preferences.rst

Reply:
@Holzhaus @Be-ing  I'm working on a better track export branch with will use grantlee for text formatting and make qobject properties generally available. This formatting function can then be changed later to use the same.

Reply:
> There is currently no demand for this. If we have a real request, we can add any info we have.

After having built this branch after a quick rebase on top of main and trying it out...

Is it possible to add a 'real request' for the track filename?

I'm currently working on providing currently playing track information from Serato and want to expand that code to support Mixxx. 

After having gone through this exercise with Serato, it becomes very clear that the number one field is the filename of the currently playing track.  With access to that, all of the metadata that Mixxx may not even have internally is now available.  It makes a huge difference as to what tooling may be written.

Thanks!

Reply:
That seems reasonable to me, although I'm not sure what metadata a separate program would want that Mixxx couldn't provide.

Reply:
For sure: Mixxx could provide it all.  But the situation gets tricky with metadata like cover art.  Now the UI will need to provide the capability to write at least two files (maybe the media has more than one embedded image?), there are image file format issues to work out, etc. As it is, a single line of output in the UI is already fairly limiting with just text output!

Outside of the filename, the metadata that is internal to Mixxx should probably be the higher priority though since there is no other source. e.g., deck, start time, # of plays, crate, etc.  If y'all want pointers as to what data has been reversed engineered out of Serato's session files, let me know!

Thanks again!

Reply:
MprisPlayer [exposes a URL to the cover art file](https://github.com/mixxxdj/mixxx/pull/3483/files?file-filters%5B%5D=.cpp#diff-9b238770d9a7f6df0f830093fed66c1dfe1b0b75342ab99d5fe71e6776d2a539R334), or if the cover art is embedded in a tag, [copies the cover art to a temporary file](https://github.com/mixxxdj/mixxx/pull/3483/files?file-filters%5B%5D=.cpp#diff-9b238770d9a7f6df0f830093fed66c1dfe1b0b75342ab99d5fe71e6776d2a539R417). We could use a similar hack for a NowPlaying.png file or something.

Alternatively we may consider building Mixxx with DBus on macOS and Windows so MPRIS could be used on those OSes, but I'm not sure how feasible that is. It might not be worth the effort.